tests/selftest.rs

```diff
@@ -37,4 +37,13 @@ fn metadata_deps() {
     assert_eq!(this.targets[1].name, "selftest");
     assert_eq!(this.targets[1].kind[0], "test");
     assert_eq!(this.targets[1].crate_types[0], "bin");
+
+    let dependencies = &this.dependencies;
+
+    let serde = dependencies
+        .iter()
+        .find(|dep| dep.name == "serde")
+        .expect("Did not find serde dependency");
+
+    assert_eq!(serde.kind, cargo_metadata::DependencyKind::Normal);
```

Comment:
I think you should also add a test for the version. Something like checking if the serde dependency has a version `> 1.0.0`.

Reply:
We currently demand that `serde` should be `"1.0.2"` - i.e. `^1.0.2 := >=1.0.2 <2.0.0`

I've added checks for the upper and lower bounds, and that a legal value is accepted.

Of course, this will break if we ever update to semver 2.0.0, but I suspect a semver 2.0.0 is unlikely (and easy to fix if so).